File: backend/ISISNotesBackend.Core/Models/NoteTextContent.cs
using ISISNotesBackend.Core.Models.Enums;

namespace ISISNotesBackend.Core.Models
{
    public class NoteTextContent : INoteContent
    {
        public string NoteId { get; set; }
        public string Type { get; set; }
        public string Text { get; set; }
        
        public NoteTextContent(string noteId, string type, string text)
        {
            NoteId = noteId;
            Type = type;
            Text = text;
        }
    }
}

File: frontend/src/app/shared/models/note-text-content.ts
import { NoteContent } from './note-content';

/**
 * Текстовый фрагмент заметки
 */
export interface NoteTextContent extends NoteContent {
  /**
   * Текстовое содержимое фрагмента заметки
   */
  text: string;
}
